eos-paygd/main.c

```diff
@@ -35,10 +35,51 @@
 #include <sys/un.h>
 #include <libeos-payg/efi.h>
 
-#define LOG_FILE_NAME "/var/log/eos-payg.log"
+#define LOGFILE_DIRNAME "/var/log/eos-payg"
+#define LOGFILE_BASENAME "eos-paygd"
+#define LOGFILE_EXT "log"
 
 #define TIMEOUT_POWEROFF_ON_ERROR_MINUTES 20
 
+static gboolean
+open_log_file (GIOChannel **log_io)
+{
+  int system_ret;
+  const gchar *log_file_name = NULL;
+  g_autoptr(GFile) log_file = NULL;
+  g_autoptr(GDateTime) now = NULL;
+  g_autofree gchar *tstamp = NULL;
+  g_autoptr(GError) error = NULL;
+
+  /* Use /bin/mkdir instead of mkdir() to ensure the mode is unaffected by
+   * the process's umask.
+   */
+  system_ret = system ("/bin/mkdir -p --mode=755 " LOGFILE_DIRNAME);
+  if (system_ret == -1 || !WIFEXITED (system_ret) || WEXITSTATUS (system_ret) != 0)
+    {
+      g_warning ("mkdir of %s failed", LOGFILE_DIRNAME);
+      return FALSE;
+    }
+
+  /* Build log file name with a date stamp so we get one file per day,
+   * ex., eos-paygd-20220418.log
+   */
+  now = g_date_time_new_now_local ();
+  tstamp = g_date_time_format (now, "%Y%m%d");
```

Comment:
Why not `%F`, the ISO-8601 date format (`%Y-%m-%d`)?

Reply:
Just personal preference of making it a number and shorter. I think `eos-paygd-20220418.log` reads better than `eos-paygd-2022-04-18.log` or `eos-paygd_2022-04-18.log` (the latter being slightly better, but I still prefer the current format.